### PR TITLE
feat: use tsconfck instead of tsconfig-resolver

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

`tsconfig-resolver` is unmaintained and doesn't support all of newer TypeScript's features. `tsconfck` is also in use by many reputable project and should be a safer dep to have.

Fix https://github.com/withastro/astro/issues/6918

## Testing

Tests should pass!

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/cli/add/index.ts [hunk 1]
```diff
@@ -848,25 +848,31 @@ async function updateTSConfig(
...
+		inputConfig = {
+			tsconfig: defaultTSConfig,
+			tsconfigFile: path.join(cwd, 'tsconfig.json'),
+			rawConfig: { tsconfig: defaultTSConfig, tsconfigFile: path.join(cwd, 'tsconfig.json') },
+		};
+	} else {
+		inputConfigText = JSON.stringify(inputConfig.rawConfig.tsconfig, null, 2);
 	}
 
+	const configFileName = inputConfig.tsconfigFile.split('/').pop();
```
Comment (bluwy) [inline packages/astro/src/cli/add/index.ts:867]:
```suggestion
	const configFileName = inputConfig.tsconfigFile.split(path.sep).pop();
```

Looks like the bug existed before, but we might need to use proper OS separator, or use `path.basename` directly.

**Review by bluwy — APPROVED** [review]

**Review by natemoo-re — APPROVED** [review]
LGTM. Thanks for pinning the exact pre-release version. We'll need to remember to keep an eye on that when it gets bumped to `latest`.

**Comment by firxworx** [comment]
Looks good this is a nice contribution @Princesseuh :)
If the motivation is to support ts5 extends format / multiple extends do you think there should be a test / fixture that covers that case? Maybe I missed when I skimmed down files changed but I didn't catch. Thanks again!

**Comment by Princesseuh** [comment]
> Looks good this is a nice contribution @Princesseuh :)
> If the motivation is to support ts5 extends format / multiple extends do you think there should be a test / fixture that covers that case? Maybe I missed when I skimmed down files changed but I didn't catch. Thanks again!

We could for sure, but we'd rather avoid testing every feature like this that are ultimately outside of our code. We trust `tsconfck` to do what it says, especially considering it's used upstream from us (Vite uses it)

**Pull request merged**